youtube_dl/extractor/rai.py

```diff
@@ -500,3 +502,108 @@ def _real_extract(self, url):
+        self._level += 1
+
+    def handle_endtag(self, tag):
+        self._level -= 1
+
+
+class RaiPlayRadioBaseIE(InfoExtractor):
+    _BASE = 'https://www.raiplayradio.it'
+
+    def parse_list(self, webpage):
+        parser = HTMLListAttrsParser()
+        parser.feed(webpage)
+        parser.close()
+        return parser.items
+
+    def get_playlist_iter(self, url):
+        webpage = self._download_webpage(url, url)
+        for attrs in self.parse_list(webpage):
+            title = attrs['data-title'].strip()
+            webpage = urljoin(url, attrs['data-mediapolis'])
+            audio_url = compat_str(self._request_webpage(
+                webpage, title).geturl())
+            yield {
+                'url': audio_url,
+                'id': attrs['data-uniquename'].lstrip('ContentItem-'),
+                'title': title,
+                'ext': determine_ext(audio_url),
+                'thumbnail': urljoin(url, attrs['data-image']),
+                'language': 'it'}
+
+    def get_playlist(self, *args, **kwargs):
+        return list(self.get_playlist_iter(*args, **kwargs))
+
+
+class RaiPlayRadioIE(RaiPlayRadioBaseIE):
+    _VALID_URL = r'%s/audio/.+?-(?P<id>%s)\.html' % (
+        RaiPlayRadioBaseIE._BASE, RaiBaseIE._UUID_RE)
+    _TEST = {
+        'url': 'https://www.raiplayradio.it/audio/2019/07/'
+               'RADIO3---LEZIONI-DI-MUSICA-'
+               '36b099ff-4123-4443-9bf9-38e43ef5e025.html',
+        'info_dict': {
+            'id': '36b099ff-4123-4443-9bf9-38e43ef5e025',
+            'ext': 'mp3',
+            'title': 'Dal "Chiaro di luna" al  "Clair de lune", '
+                     'prima parte con Giovanni Bietti',
+            'thumbnail': 'https://www.raiplayradio.it/cropgd/400x400/dl/'
+                         'img/2019/07/08/1562590329054_luna.jpg',
+            'language': 'it'}}
+
+    def _real_extract(self, url):
+        list_url = url.replace('.html', '-list.html')
+        for entry in self.get_playlist_iter(list_url):
+            if entry['id'] == self._match_id(url):
+                return entry
+
+
+class RaiPlayRadioPlaylistIE(RaiPlayRadioBaseIE):
+    _VALID_URL = r'%s/playlist/.+?-(?P<id>%s)\.html' % (
+        RaiPlayRadioBaseIE._BASE, RaiBaseIE._UUID_RE)
+    _TEST = {
+        'url': 'https://www.raiplayradio.it/playlist/2017/12/'
+               'Alice-nel-paese-delle-meraviglie-'
+               '72371d3c-d998-49f3-8860-d168cfdf4966.html',
+        'info_dict': {
+            'id': '72371d3c-d998-49f3-8860-d168cfdf4966',
+            'title': "Alice nel paese delle meraviglie",
+            'description': "di Lewis Carrol letto da Aldo Busi"},
+        'playlist_count': 11}
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        playlist_webpage = self._download_webpage(url, playlist_id)
+        playlist_title = unescapeHTML(self._html_search_regex(
+            r'data-playlist-title="(.+?)"', playlist_webpage, 'title'))
```

Comment:
Playlist title is optional.

Reply:
Fixed: it will not be added to the results if missing.